server/routes/api/urls/urls.ts

```diff
@@ -53,7 +53,11 @@ router.post(
       authorize(actor, "read", user);
       authorize(actor, "read", document);
 
-      ctx.body = await presentMention(user, document);
+      ctx.body = await presentUnfurl({
```

Comment:
I'm not sure if moving these into a single presenter is beneficial, you've lost type safety here that would be retained simply by using `presentMention` directly?

Reply:
Well, the long-term goal is to eliminate need for treating "unfurling" separately for mentions, documents and external resources. The following pattern seems to be emerging:

- Parse URL
- Fetch resource(Database/REST)
- Present

Collating into a single presenter was an initial step.

Reply:
> Well, the long-term goal is to eliminate need for treating "unfurling" separately for mentions, documents and external resources.

True enough,